mail/api.py

```diff
@@ -114,9 +126,14 @@ def send_batch(cls, subject, body, recipients,  # pylint: disable=too-many-argum
             chunk_size (int): The maximum amount of emails to be sent at the same time
 
         Returns:
-            list: List of requests.Response HTTP response from Mailgun
+            list:
+                List of (recipients, requests.Response, exception) where
+                recipients is a list of recipient emails,
+                Response is an HTTP response from Mailgun, if there is one,
+                exception is an exception which occurred when attempting to mail, if there is one
```

Comment:
The Unix philosophy states that no feedback means that everything was successful. As a result, I would suggest that this function have the following return values:

* Every request to Mailgun was successful: return a list of response objects, just as this function used to do. The caller can assume that every email address in the recipients list was processed correctly.
* At least one request to Mailgun was unsuccessful: raise an exception. Before the exception is raised, however, construct the exception object and attach the list of response objects to it, as well as the list of email addresses that were processed correctly. The caller can catch this exception if this information is necessary.

I'm very, very wary about returning error values instead of raising exceptions -- you're expecting the caller to check for the error value, and it's easy to forget to do so. If we change the code to work the way I've described, there's no chance for a failure to pass unnoticed. The caller has to intentionally handle the exception, or else let it bubble up.